ast/field.go

```diff
@@ -430,6 +447,10 @@ func (n *SyntheticOneOf) TrailingComments() []Comment {
 	return nil
 }
 
+func (n *SyntheticOneOf) OneOfName() Node {
```

Comment:
The `SyntheticOneOf` struct is already implemented, so I didn't need to apply all of these changes from the upstream: https://github.com/jhump/protoreflect/pull/512/files#diff-cb77b26bf20edef5ffaa76b833566cc54c2dbad653dc1cc6f0e9d1868d276986R392-R397

Reply:
Interesting. I see the [commit where I added it](https://github.com/bufbuild/protocompile/commit/7df11bfc51cabc51515cc67b22997019a6ce0755), but for the life of me I can't remember _why_ I added it then. That was over a year ago -- back in the early "hack and slash" times of getting this repo into shape :)

One significant different I see is that the `protoparse` version refers to the proto3 optional _field_, not just the name/identifier. That could be useful change, just to make the provenance of the synthetic node more clear.

```diff
 type SyntheticOneOf struct {
-	Ident IdentValueNode
+	Field *FieldNode
 }
```

Reply:
Updated - can you take another look?